### PR TITLE
Added action events

### PR DESCRIPTION
**Pull request by ankitgoyalgloballogic**
#REF: #MOBILEAPPS-1229

### CODE REVIEW

File: ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift [hunk 1]
```diff
@@ -154,3 +154,41 @@ extension ListComponentViewController: NodeActionsViewModelDelegate,
...
+
+// MARK: - Analytics
+extension ListComponentViewController {
+    
+    func logEvent(with action: ActionMenu?, node: ListNode?) {
+        guard let action = action else { return }
+        
+        switch action.type {
+        case .addFavorite:
+            AnalyticsManager.shared.fileActionEvent(for: node, eventName: .removeFromFavorites)
```
Comment (aman-alfresco) [inline ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift:166]: its action is addFavorite but eventName is removeFromFavorites.

Similar case for removeFavorite action

File: ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift [hunk 2]
```diff
@@ -154,3 +154,41 @@ extension ListComponentViewController: NodeActionsViewModelDelegate,
...
+        case .removeFavorite:
+            AnalyticsManager.shared.fileActionEvent(for: node, eventName: .addToFavorites)
+        case .moveTrash:
+            AnalyticsManager.shared.fileActionEvent(for: node, eventName: .moveToTrash)
+        case .restore:
+            AnalyticsManager.shared.fileActionEvent(for: node, eventName: .restoreFromTrash)
+        case .permanentlyDelete:
+            AnalyticsManager.shared.fileActionEvent(for: node, eventName: .permanentlyDelete)
+        case .moveToFolder:
+            AnalyticsManager.shared.fileActionEvent(for: node, eventName: .move)
```
Comment (aman-alfresco) [inline ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift:176]: remove cases and pass action as event name

File: ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift (hunk 1 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift]: Done

File: ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/Components/ListComponent/ListComponent+Actions.swift]: Done

**Review by aman-alfresco — APPROVED** [review]

**Pull request merged**